### DISCUSSION TITLE
IOS : RangeError: Maximum call stack size exceeded.

### DISCUSSION THREAD

**Discussion by dingsq820** [WalletConnect/walletconnect-monorepo#2259] [Developer Support] [answered]
version:
`"@walletconnect/web3wallet": "^1.5.1"
@walletconnect/sign-client": "^2.6.2"`
sdk:Web3Wallet SDK
code:`core.pairing.pair({ uri: wcuri })`

When pair to the dapp,an error was reported on IOS. Android is fine.
`ERROR  {"context": "core"} {"context": "core/relayer"} [RangeError: Maximum call stack size exceeded.]`

**Comment by dingsq820** [comment]
It seems that if the [walletconnect/core] package is v2.7.1, this error will be reported.

**Comment by finessevanes** [comment]
Hey - are you able to share the implementation?

**Comment by dingsq820** [comment, marked as answer]
https://github.com/WalletConnect/react-native-examples/tree/main/wallets/rn_cli_wallet_068_5
Removing【BigInt】 from the following code provided in the example will not report this error

```
const TextEncodingPolyfill = require('text-encoding');
const BigInt = require('big-integer');

Object.assign(global, {
  TextEncoder: TextEncodingPolyfill.TextEncoder,
  TextDecoder: TextEncodingPolyfill.TextDecoder,
  BigInt,
});
```

**Comment by dingsq820** [comment]
@finessevanes 
Follow this example to achieve
https://github.com/WalletConnect/react-native-examples/tree/main/wallets/rn_cli_wallet_068_5

**Reply by hyrbi** [reply]
Hi, @dingsq820! Can you please share the solution? I just tried to delete part of the code about BigInt, but it doesn't help me. Maybe you can fork the example rep and show it?
